odoo/tools/mail.py

```diff
@@ -635,3 +636,15 @@ def encapsulate_email(old_email, new_email):
         name_part,
         new_email_split[0][1],
     ))
+
+
+def url_domain_extract(url):
+    """ Extract the company domain to be used by IAP services notably. Domain
+    is extracted from an URL e.g:
+        - www.info.proximus.be -> proximus.be
```

Comment:
Not all ccTLD allow unrestricted SLDs, this is going to screw up tremendously for .uk, .br, .au, ... and some countries allow SLDs but restrict some uses to 3LDs e.g. companies, governmental services, educational institutions, ... in Japan are restricted to 3LDs under known SLDs, and geographical locations have their own scheme which goes down to 4LD.

It also doesn't work for our own non-premise clients e.g. saas/paas, as they're all 3LDs under `odoo.com`.

This function would probably be better off being removed entirely.

Reply:
This part is actually a backport (suggested by @awa-odoo) from a commit in PR #64600 so I think we'll have to change that in latest versions too

Reply:
My proposal is to just not backport it at all and not use it.

Reply:
Did not find the utility ... maybe lost in another discussion ?